### PR TITLE
Enable rgbmatrix on IDF 5.1

### LINKED ISSUE #8423: Re-enable `rgbmatrix` on ESP
Protomatter probably needs to be updated to ESP-IDF 5.

### PR DESCRIPTION
**Pull request by tannewt**
Fixes #8423

### CODE REVIEW

File: shared-bindings/framebufferio/FramebufferDisplay.c [hunk 1]
```diff
@@ -153,7 +162,14 @@ STATIC mp_obj_t framebufferio_framebufferdisplay_obj_refresh(size_t n_args, cons
     if (minimum_frames_per_second > 0) {
         maximum_ms_per_real_frame = 1000 / minimum_frames_per_second;
     }
-    return mp_obj_new_bool(common_hal_framebufferio_framebufferdisplay_refresh(self, 1000 / args[ARG_target_frames_per_second].u_int, maximum_ms_per_real_frame));
+
+    uint32_t target_ms_per_frame;
+    if (args[ARG_target_frames_per_second].u_obj == mp_const_none) {
+        target_ms_per_frame = 0xffffffff;
```
Comment (dhalbert) [inline shared-bindings/framebufferio/FramebufferDisplay.c:168]: Could you `#define` this sentinel value with a descriptive name in one of the .h files?

File: shared-module/framebufferio/FramebufferDisplay.c [hunk 2]
```diff
@@ -290,7 +290,7 @@ uint16_t common_hal_framebufferio_framebufferdisplay_get_rotation(framebufferio_
 
 
 bool common_hal_framebufferio_framebufferdisplay_refresh(framebufferio_framebufferdisplay_obj_t *self, uint32_t target_ms_per_frame, uint32_t maximum_ms_per_real_frame) {
-    if (!self->auto_refresh && !self->first_manual_refresh) {
+    if (!self->auto_refresh && !self->first_manual_refresh && (target_ms_per_frame != 0xffffffff)) {
```
Comment (dhalbert) [inline shared-module/framebufferio/FramebufferDisplay.c:293]: Use sentinel value name instead of magic number.

**Review by dhalbert — CHANGES_REQUESTED** [review]
Looks good - one style request.

File: shared-bindings/framebufferio/FramebufferDisplay.c (hunk 1 above)
Comment (tannewt) [reply shared-bindings/framebufferio/FramebufferDisplay.c]: Added `NO_FPS_LIMIT`.

**Review by dhalbert — APPROVED** [review]
Thanks!

**Review by dhalbert — CHANGES_REQUESTED** [review]
Hmm, build failures - not sure what is different from before.

**Comment by tannewt** [comment]
It didn't build before either. I pushed a fix but will need to PR it to protomatter first before this can be merged. Using the CI here to make sure everything is fixed.

**Comment by FoamyGuy** [comment]
I tested the matrixportal_s3 artifact from this PR successfully on hardware with a somewhat basic displayio test script. It does allow for  successful import of `rgbmatrix` and seems to be working as expected with my test script.

**Review by dhalbert — APPROVED** [review]
Thanks!

**Pull request merged**